### PR TITLE
First ECAL implementation

### PR DESCRIPTION
**Pull request by kharlov**

### CODE REVIEW

**Comment by njacazio** [comment]
Hi @kharlov please trigger me when you need to have this PR reviewed

**Comment by kharlov** [comment]
Hi @njacazio , I fixed several bugs in ECAL response, please check and merge the new PR.

File: examples/aod/createO2tables.C [hunk 1]
```diff
@@ -32,18 +32,19 @@ R__LOAD_LIBRARY(libDelphesO2)
...
+const double Bz = 5.e-1;
 // TOF
-const double tof_radius = 100.; // [cm] Radius of the TOF detector (used to compute acceptance)
-const double tof_length = 200.; // [cm] Length of the TOF detector (used to compute acceptance)
-const double tof_sigmat = 0.02; // [ns] Resolution of the TOF detector
-const double tof_sigmat0 = 0.2; // [ns] Time spread of the vertex
+const double tof_radius = 100.;
+const double tof_length = 200.;
+const double tof_sigmat = 0.020;
+const double tof_sigmat0 = 0.20;
```
Comment (njacazio) [inline examples/aod/createO2tables.C]: This should not need to be changed

File: examples/aod/createO2tables.C [hunk 2]
```diff
@@ -70,7 +71,7 @@ const char* inputFileAccMuonPID = "muonAccEffPID.root";
 
 // Simulation parameters
 constexpr bool do_vertexing = true;  // Vertexing with the O2
-constexpr bool enable_nuclei = true; // Nuclei LUTs
+constexpr bool enable_nuclei = true;
```
Comment (njacazio) [inline examples/aod/createO2tables.C:75]: This should not need to be changed

File: examples/aod/createO2tables.C [hunk 3]
```diff
@@ -270,6 +275,25 @@ int createO2tables(const char* inputFile = "delphes.root",
         dNdEta += 1.f;
       }
       FillTree(kMcParticle);
+
+      // info for the ECAL
+      float posZ, posPhi;
+      TLorentzVector pECAL;
+      if (ecal_detector.makeSignal(*particle, pECAL, posZ, posPhi)) { // to be updated 13.09.2021
```
Comment (njacazio) [inline examples/aod/createO2tables.C:297]: The initalization of the TLorentzVector could be done once and used outside of the loop for tracks

**Review by njacazio — CHANGES_REQUESTED** [review]
Hi @kharlov  good work! Could you also run `clang-format -i` on the files that you changed so as to uniformize the style?
If it does not work it should be possible to do it from inside the alienv environment

**Comment by njacazio** [comment]
Also, there are some conflicts that have to solved before this can be merged

**Comment by kharlov** [comment]
Hi @njacazio , I solved conflicts, ran clang and opened a new PR. Did I do it correctly?

**Comment by njacazio** [comment]
Hi Yuri, yes, I will have a look asap

**Comment by njacazio** [comment]
Hi @kharlov your changes will be merged from here https://github.com/AliceO2Group/DelphesO2/pull/126

**Comment by kharlov** [comment]
Hi @njacazio , I fixed one more bug in ECAL response: photon hit coordinate was not calculated correctly.

**Comment by njacazio** [comment]
Hi, @kharlov, perfect, I will include it in the software as soon as you push it in this PR

**Comment by njacazio** [comment]
Ah, I just saw that you already did, I will import it now

**Comment by njacazio** [comment]
Closing as already merged in other PR

**Pull request closed without merging**